libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -846,23 +806,46 @@ DNNReturnType ff_dnn_execute_model_async_ov(const DNNModel *model, DNNExecBasePa
         return DNN_ERROR;
     }
 
-    while (ff_queue_size(ov_model->inference_queue) >= ctx->options.batch_size) {
+    if (ctx->options.async) {
+        while (ff_queue_size(ov_model->inference_queue) >= ctx->options.batch_size) {
+            request = ff_safe_queue_pop_front(ov_model->request_queue);
+            if (!request) {
+                av_log(ctx, AV_LOG_ERROR, "unable to get infer request.\n");
+                return DNN_ERROR;
+            }
+
+            ret = execute_model_ov(request, ov_model->inference_queue);
+            if (ret != DNN_SUCCESS) {
+                return ret;
+            }
+        }
+
+        return DNN_SUCCESS;
+    }
+    else {
+        if (model->func_type == DFT_ANALYTICS_CLASSIFY) {
+            // Once we add async support for tensorflow backend and native backend,
+            // we'll combine the two sync/async functions in dnn_interface.h to
```

Comment:
not catch this point.

we do support asyn for tf backend now.
and sync/async is combined in next patch without any change here.

I know that classification is not supported/tested for sync in the previous, is this the root cause? So we return error here to first keep the same feature as previous.

Reply:
Yes. I checked with the following two filters in this way and found them to be working with sync mode as well.

```sh
ffmpeg -i cici.jpg -vf dnn_classify=dnn_backend=openvino:model=emotions-recognition-retail-0003.xml:input=data:output=prob_emotion:confidence=0.3:labels=emotions-recognition-retail-0003.label:target=face:backend_configs=async=0,showinfo -f null -
```

```sh
ffmpeg -i cici.jpg -vf dnn_detect=dnn_backend=openvino:model=face-detection-adas-0001.xml:input=data:output=detection_out:confidence=0.6:labels=face-detection-adas-0001.label:backend_configs=async=0,showinfo -f null -
```

But I did not remove these lines as I wasn't sure if it will work in all cases. Like if there are other cases as well.

Reply:
the first command line is not correct, dnn_classify will do nothing since no detect result, and the second command line just contain dnn_detect, not dnn_classify.
To use classify, we need to use the two filters together:  -vf dnn_detect=...dnn_classify=...

my concern was about the comment, we may update the comment to be more clear, keeping return error.

Reply:
Oh yes, I just noticed it didn't return any coordinates of the regions classified.

How about this?
```
        // Classification filter has not been completely tested with the sync
        // mode on the TensorFlow and Native backend. So, do not support now
```

Reply:
just the first line is enough.

we can always support here no matter if tf/native backends are supported or not,

Reply:
Okay. Updated.